jsonschema/tests/test_validators.py

```diff
@@ -898,6 +921,77 @@ def test_helpful_error_message_on_failed_pop_scope(self):
             resolver.pop_scope()
         self.assertIn("Failed to pop the scope", str(exc.exception))
 
+    def test_prev_resolve_fragment_raises_exception(self):
+        schema = {
+            "definitions": {
+                "Pet": {
+                    "properties": {
+                        "category": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/Category"
+                            }
+                        },
+                        "name": {
+                            "type": "string"
+                        }
+                    }
+                },
+                "Category": {
+                    "required": [
+                        "name"
+                    ],
+                    "properties": {
+                        "name": {
+                            "type": "string"
+                        },
+                        "id": {
+                            "type": "integer",
+                            "format": "int32"
+                        }
+                    }
+                }
+            }
+        }
+        resolver = RefResolver.from_schema(schema)
+        resolver.resolve_fragment = mock.MagicMock(side_effect=self.prev_resolve_fragment)
+        self.assertRaises(RefResolutionError, resolver.resolve_fragment, schema, "#/definitions/Category")
+
+    def test_resolve_fragment(self):
+        schema = {
+            "definitions": {
+                "Pet": {
+                    "properties": {
+                        "category": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/Category"
+                            }
+                        },
+                        "name": {
+                            "type": "string"
+                        }
+                    }
+                },
+                "Category": {
+                    "required": [
+                        "name"
+                    ],
+                    "properties": {
+                        "name": {
+                            "type": "string"
+                        },
+                        "id": {
+                            "type": "integer",
+                            "format": "int32"
+                        }
+                    }
+                }
+            }
+        }
+        resolver = RefResolver.from_schema(schema)
+        self.assertEqual(resolver.resolve_fragment(schema, "#/definitions/Category"), schema['definitions']['Category'])
```

Comment:
I assume that what this is trying to fix is probably validation related, so would be nice to have at least one actual integration test (that uses a relative reference in a schema for validation) -- especially so, since if there's a bug here, it means we're missing a test in the [official test suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite)